darshan-util/pydarshan/tests/test_plot_common_access_table.py

```diff
@@ -0,0 +1,229 @@
+import pytest
+
+import pandas as pd
+from pandas.testing import assert_frame_equal # type: ignore
+
+import darshan
+from darshan.experimental.plots import plot_common_access_table
+
+
+@pytest.mark.parametrize("report, mod, expected_df",
+    [
+        (
+            darshan.DarshanReport("./examples/example-logs/ior_hdf5_example.darshan"),
+            "POSIX",
+            pd.DataFrame(
+                data=[[262144, 32], [512, 9], [544, 5], [328, 3]],
+                columns=["Access Size", "Count"],
+            ),
+        ),
+        (
+            darshan.DarshanReport("./examples/example-logs/ior_hdf5_example.darshan"),
+            "MPI-IO",
+            pd.DataFrame(
+                data=[[262144, 32], [512, 9], [544, 5], [328, 3]],
+                columns=["Access Size", "Count"],
+            ),
+        ),
+        (
+            darshan.DarshanReport("./examples/example-logs/ior_hdf5_example.darshan"),
```

Comment:
As another example I also checked `nonmpi_partial_modules.darshan`:

![image](https://user-images.githubusercontent.com/7903078/148326195-92ef1a37-bb7c-4fbf-a86a-5076cca2c816.png)

vs. this PR 

```
========================================
mod:  POSIX
   Access Size  Count
0         1024   7692
1           32    276
2          100    269
3           92    265
```

Anyway, I'm just saying this is the kind of analysis vs. previous implementation that can substantially boost confidence, and should be trivial to add another few cases to the suite, esp. once gh-587 is in.

Reply:
Thanks for the recommendations, these are all good points. I've moved to leveraging the log retrieval machinery now, and I've added the test case shown here. 

A quick note -- before using `get_log_path`, the test completed in ~.9 seconds, and now takes about 3.5 s on my Mac. Probably not a big deal, but if we eventually switch over dozens or more tests, this could add to the test suite run time pretty considerably.   

I

Reply:
> A quick note -- before using get_log_path, the test completed in ~.9 seconds, and now takes about 3.5 s on my Mac. Probably not a big deal, but if we eventually switch over dozens or more tests, this could add to the test suite run time pretty considerably.

For me, the latest version of this branch is faster than your originally reported time:

`pytest tests/test_plot_common_access_table.py`

`15 passed in 0.48s`

In fact, if I checkout `31f51b75`, the hash before you use the `get_log_path()` machinery it takes the same amount of time, so I'm definitely not believing the ratio you note:

`15 passed in 0.48s`

Nonetheless, if you really believe you are experiencing that kind of slowdown, feel free to carefuly help with gh-606.

Reply:
That's interesting. I checked `pytest tests/test_plot_common_access_table.py` again and I'm still seeing much slower times. 

```                          
======================================================================== test session starts =========================================================================
platform darwin -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /Users/nawtrey/darshan_install/darshan/darshan-util/pydarshan, configfile: setup.cfg
plugins: xdist-2.4.0, forked-1.3.0, cov-3.0.0
collected 15 items                                                                                                                                                   

tests/test_plot_common_access_table.py ...............                                                                                                         [100%]

========================================================================= 15 passed in 3.75s =========================================================================
```

But yeah, probably not a concern until we start simplifying some of the other tests. If the problem persists I'll have to dig deeper, for now I just wanted to point out my observation.

Reply:
Maybe repeat your analysis on some Linux machines as wel (`darwin` node maybe), including the before/after commit hashes, and post it in the cross-linked issue for me to inspect.

Reply:
Also, probably best to report the results on your mac and on linux with i.e., `--durations=20` so I can see the actual timings before/after like this:

```
================================================================================================================================ slowest 20 durations =================================================================================================================================
55.05s call     tests/test_summary.py::test_main_all_logs_repo_files
5.14s call     tests/test_report.py::test_jobid_type_all_logs_repo_files
3.43s call     tests/test_summary.py::test_main_without_args[argv4-2]
2.61s call     tests/test_summary.py::TestReportData::test_stylesheet[tests/input/sample-dxt-simple.darshan]
2.50s call     tests/test_summary.py::TestReportData::test_header_and_footer[examples/example-logs/dxt.darshan-N/A (2020-04-21)]
2.46s call     tests/test_summary.py::TestReportData::test_header_and_footer[tests/input/sample-dxt-simple.darshan-a.out (2021-04-22)]
2.43s call     tests/test_summary.py::test_main_with_args[argv0]

```

Reply:
Okay, it looks like the issue here was I had a bunch of `.darshan` files in `/benchmarks/env/`, `.tox`, and `/build`, totaling 243 darshan files. The correct number is 14, so this was substantially slower. I think there are still some unnecessary log paths being added, so maybe we just want to restrict the glob search a bit more in `log_utils._locate_log` to prevent things like this.

Reply:
That idea for improvement is already listed in the cross-listed issue, along with some others that are probably even better.

Like I said, feel free to help with the cross-listed issue where I provide some ideas. Please check performance enhancements carefully for correctness of course.

Reply:
Also, remember the broader plan to install the file assets and *not* search the local source repo at all (only the installed version, which shouldn't have your local junk). This is important, because on HPC installs, there is no local version.

In fact, it sounds like you may be running the tests with an in-source import instead of using i.e., `--import-mode=importlib `, which shouldn't have the extra stuff either really.